src/cache.jl

```diff
@@ -94,7 +94,11 @@ end
 updatecache(absentmodule::Symbol, server) = updatecache([absentmodule], server)
 
 function updatecache(absentmodules::Vector{Symbol}, server)
+    server.cp_lock && return
```

Comment:
Is this how we want things to be? Essentially if you can't acquire the lock, give up? That seems not ideal... Generally, a better strategy might be to have just one background green task that waits on a ``Channel`` for requests to update the cache with a given module, and then that task starts the child process. So the normal language server code would no longer start a new child process when it needs stuff for a module, instead it would push that module name into the ``Channel``, then the background task processes the items from the channel one by one, i.e. we would never have more than one julia child process running at the same time that loads modules, but at the same time all requests would be served eventually.